packages/offix-cache/src/createMutationOptions.ts

```diff
@@ -242,18 +250,21 @@ function deleteItemFromQuery({ mutationName, updateQuery, idField = "id" }: Cach
             toBeRemoved = operationData;
           }
           let newData: any;
-          if (typeof queryResult[queryField].filter === "function") {
-            newData = queryResult[queryField].filter((item: any) => {
-              return toBeRemoved[idField] !== item[idField];
-            });
-          } else if (queryResult[queryField].items) {
-            const newItems = queryResult[queryField].items.filter((item: any) => {
+
+          const prev = (returnField)
+            ? queryResult[queryField][returnField]
+            : queryResult[queryField];
+
+          if (typeof prev.filter === "function") {
```

Comment:
When I think about it - it will be better to check if prev is array

Reply:
Okay sure, easy fix